fire_atlas_editor/lib/store/store.dart

```diff
@@ -38,10 +73,25 @@ class ModalState extends Equatable {
 }
 
 class FireAtlasState {
-  FireAtlas? currentAtlas;
+  LoadedProjectEntry? loadedProject;
   bool hasChanges = false;
   BaseSelection? selectedSelection;
   ModalState? modal;
   List<Message> messages = [];
   Rect? canvasSelection;
+
+  FireAtlas? get currentAtlas => loadedProject?.project;
+
+  void set currentAtlas(FireAtlas? atlas) {
+    if (atlas == null) {
+      throw "Can't set a null atlas";
```

Comment:
why not just not accept nullable?

Reply:
We can't, because the getter can be null, I am not 100% happy with this as well, but to throw an exception was the best I could think of right now.